csrc/multidevice/utils.cpp

```diff
@@ -317,30 +317,52 @@ void insertReshardingsAfter(Fusion* fusion) {
   }
 }
 
+void setShardedAllocationDomain(TensorView* tv) {
+  if (!tv->hasAllocation()) {
+    tv->setAllocationDomain(tv->getLeafDomain(), tv->getContiguity());
+  }
+}
+
 } // namespace
 
-void propagateShardings(Fusion* fusion) {
+void propagateShardingsAndSetAllocationDomain(Fusion* fusion) {
+  // Set global input's allocation domain.
+  // Currently we only propagate shardings from consumer to producer,
+  // so a DeviceMesh is required.
+  for (auto global_input_tv :
+       ir_utils::filterByType<TensorView>(fusion->inputs())) {
+    NVF_ERROR(
+        global_input_tv->hasDeviceMesh(),
+        "Global inputs must be assigned a DeviceMesh ",
+        global_input_tv->toString());
+    setShardedAllocationDomain(global_input_tv);
```

Comment:
Can you remind me of the purpose of this PR? IIRC, the problem was that certain resharding operations require some specific allocation order, e.g. the DID dimensions need to be put up front and have to be contiguous, etc. But the changes in this function are a bit overkilling? For example, I don't yet understand why the complete fusion's inputs need to have the same allocation domain as the leaf domain, if they are not resharded immediately.

Reply:
FYI: I don't think we need to set this on inputs, *currently*, propagation only goes from inputs to outputs. but that might change in the future where we would set intermediates as we see fit.
Nevertheless, explicitly marking input might not be necessary, since nobody is supposed to change this.

Reply:
Oops, my bad I didn't need to set _all_ the sharded tensor's allocation. It has been updated to just the one's around resharding expressions!